tests/integration/test_reinstall.py

```diff
@@ -299,3 +300,89 @@ async def test_rsync_fail(one_src, one_run, mock_glbl_cfg, non_interactive):
     with pytest.raises(WorkflowFilesError) as exc_ctx:
         await reinstall_cli(opts=ReInstallOptions(), workflow_id=one_run.id)
     assert 'An error occurred reinstalling' in str(exc_ctx.value)
+
+
+@pytest.fixture
+def my_install_plugin(monkeypatch):
+    """This configures a single post_install plugin.
+
+    The plugin starts an async task, then returns.
+    """
+    progress = []
+
+    @EntryPointWrapper
+    def post_install_basic(*_, **__):
+        """Simple plugin that returns one env var and one template var."""
+        nonlocal progress
+
+        async def my_async():
+            # the async task
+            nonlocal progress
+            await asyncio.sleep(2)
+            progress.append('end')
+
+        # start the async task
+        progress.append('start')
+        asyncio.get_event_loop().create_task(my_async())
```

Comment:
According to the docs you should assign this to something: https://docs.python.org/3/library/asyncio-task.html#asyncio.create_task

Reply:
Hmm... This is just the test which proves the concept, not functional code, so this isn't an issue.

But it is what Rose is doing at the other end, so you might want to stick this comment here: https://github.com/metomi/rose/pull/2744/files#diff-8db25a5509044308d363a425af008c5ba353b7d04f1d78dc08c3a4c5a659afb9R188

This doesn't appear to be an issue, but we should ensure that it can't become one.

The tasks created by Rose *are* referenced because they get added to a list called `running`:

https://github.com/metomi/rose/blob/191ce20b8515e9624a7e9081071a8b6533da770e/metomi/rose/job_runner.py#L223-L227

However, the task which creates those tasks is not referenced:

https://github.com/metomi/rose/blob/191ce20b8515e9624a7e9081071a8b6533da770e/metomi/rose/job_runner.py#L188

Simplified call stack:

* cylc-flow (async)
  * cylc-rose (**sync**)
    * metomi-rose (**sync**)
      * _run (async) - **un**referenced
        * task (async) - referenced 

So I would need to find a way to ensure that task remains referenced. It would be possible to pass it back through the callchain which would remove the need for the `async_block`, but would get messy in other ways.

Reply:
Addressed in the metomi-rose PR: https://github.com/metomi/rose/pull/2744/files#r1473137433